_posts/2024-03-20-infinispan-15-tracing.adoc

```diff
@@ -0,0 +1,135 @@
+---
+layout: blog
+title: Infinispan 15 Tracing Improvement
+permalink: /blog/:year/:month/:day/infinispan-15-tracing
+date: '2024-03-20T9:00:00.000-00:00'
+author: fax4ever
+tags: [ "opentelemetry", "tracing" ]
+---
+
+= Infinispan 15 tracing news
+
+In the previous version of Infinispan https://infinispan.org/blog/2022/07/18/infinispan-14-opentelemetry-tracing[we introduced the OpenTelemetry Tracing capability].
+With the new version we evolved the integration in order to:
+
+* Allow to configure it both globally and on-cache basis using the standard Infinispan configuration.
+* Trace more event types: cluster, cross-site, cache-store and security.
+* Allow to select and configure, also at runtime, the span types that we want to produce and send to the collector.
+
+All the source code we're presenting on this article is taken from https://github.com/fax4ever/infinispan-demo/tree/main/infinispan-15-tracing[this demo project].
+
+== Configure tracing
+
+=== Global configuration
+
+To enable tracing globally you need as usual to define the `collector-endpoint`, but instead of using a system property
+it is possible to use the infinispan server configuration, in which we introduced a component to configure the tracing globally.
+
+[source,xml]
+----
+<infinispan xmlns:server="urn:infinispan:server:15.0">
+   <cache-container name="default">
+      <tracing collector-endpoint="http://jaeger-collector.infinispan-15-tracing.svc.cluster.local:4318" security="true" service-name="infinispan-service-tracing" />
+   </cache-container>
+</infinispan>
+----
+
+By default we support the tracing with the OpenTelemetry Protocol (OTLP),
+with the default exporter protocol `http/protobuf`, and not anymore `grpc`.
+For instance the default Jaeger port using this combination is the `4318`.
+
+We also activate the tracing of the security audit. This is a new tracing category we introduced with Infinispan 15 and it is defined at cache manager level (globally).
+In the demo this is the configuration passed to the Infinispan Helm Chart:
+
+[source,yaml]
+----
+infinispan:
+ cacheContainer:
+   tracing:
+     collector-endpoint: "http://jaeger-collector.infinispan-15-tracing.svc.cluster.local:4318"
+     service-name: infinispan-service-tracing
+     security: true
+----
+
+=== Cache configuration
+
+The other tracing categories are configurable at cache level, since they are cache scoped.
```

Comment:
Cache scoped tracing categories must be configured in the cache configuration.
Here's an example of a cache configuration that enables tracing for additional categories: